### PR TITLE
SVN support

### PR DESCRIPTION
**Pull request by suryagaddipati**
Had to add Svn support to make it work with our repos at work

### CODE REVIEW

**Comment by srushti** [comment]
Thanks, but I don't think we can decide on the scm being used based on the beginning of the url. We can't be sure that git projects are using the git protocol, or that the user used in ssh is git. It could be added over http or the user for ssh could be something else.
We'll need some other mechanism to branch the logic.

File: app/models/scm_provider.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      end
+      def version(file_path)
+        "svn info #{file_path}"
+      end
+    end
+  end
+
+
+  def self.provider(url)
+    url.starts_with?("git")? ScmProvider::Git : ScmProvider::Svn
```
Comment (kaiwren) [inline app/models/scm_provider.rb:45]: @suryagaddipati Nice patch. Minor thing - the autodetection could barf if you're using an ssh or http based git repo. I'd default to Git, not the other way around. In fact, this probably needs to bubble all the way up to a command line option allowing the user to explicitly set an SCM.
WDYT @srushti?

Comment (srushti) [inline app/models/scm_provider.rb:45]: Yup, that's exactly what I said in the earlier comment We can't decided the scm based on starts_with. You could look at how branch is being handled currently, you probably want to do something like that. And, I agree the default should definitely remain git.

**Comment by suryagaddipati** [comment]
@srushti Agree the implementation was pretty naive to assume that git repos start with 'git'. was mostly meant as first pass.
@kaiwren yes there is no foolproof way to guess what kind of scm it is from the url. I agree that git should be the default if no scm is specified while adding the repo. Will  make that change.

**Comment by kaiwren** [comment]
Let me know if you're pressed for time - I'm happy to merge your changes to my local branch and proceed with adding the CLI components.

**Comment by suryagaddipati** [comment]
@kaiwren thanks I had made some progress on it i will should be done with it in a few.

**Pull request closed without merging**